sepal_ui/sepalwidgets/sepalwidget.py

```diff
@@ -17,25 +17,57 @@ class SepalWidget(v.VuetifyWidget):
     viz = Bool(True).tag(sync=True)
     "Bool: whether the widget is displayed or not"
 
-    old_class = ""
-    "str: a saving attribute of the widget class"
+    old_class = Unicode("").tag(sync=True)
+    "Unicode: a saving attribute of the widget class"
+
+    def __init__(self, **kwargs):
+
+        # remove viz from kwargs
+        # class_list need to be setup before viz
+        # to let hide and shw function run
+        viz = kwargs.pop("viz", True)
+
+        # init the widget
+        super().__init__(**kwargs)
+
+        # setup the viz status
+        self.viz = viz
 
     @observe("viz")
     def _set_viz(self, change):
         """
-        hide or show the component according to its viz param value
+        hide or show the component according to its viz param value.
+
+        Hide the widget by reducing the html class to :code:`d-none`.
+        Show the widget by removing the :code:`d-none` html class.
+        Save the previous class
 
         Args:
             change: the dict of a trait callback
         """
 
-        self.show() if change["new"] else self.hide()
+        # will be replaced byt direct calls to built-in hide
+        # once the previous custom implementation will be fully removed
+
+        if self.viz:
+
+            # change class value
+            self.class_ = self.old_class or self.class_
+            self.class_list.remove("d-none")
+
+        else:
+
+            # change class value
+            self.class_list.remove("d-none")
+            self.old_class = str(self.class_)
+            self.class_ = "d-none"
 
         return
 
+    @deprecated(version="2.6.0", reason="Equivalent to :code:`self.viz = not self.viz`")
     def toggle_viz(self):
```

Comment:
I think this method is simpler than changing the state by traits

Reply:
then lets remove the deprecation notice